xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java

```diff
@@ -474,24 +488,27 @@ public void validateExtensions() throws QueryException, XWikiException
         }
     }
 
-    public ResourceReference getDownloadReference(XWikiDocument document, BaseObject extensionVersionObject)
-    {
-        // Has a version
-        String extensionVersion = getValue(extensionVersionObject, XWikiRepositoryModel.PROP_VERSION_VERSION);
+    public ResourceReference getDownloadReference(XWikiDocument document, String extensionVersion) throws ResolveException {
 
-        // The download reference seems ok
-        String download = getValue(extensionVersionObject, XWikiRepositoryModel.PROP_VERSION_DOWNLOAD);
+        String downloadURL = null;
+
+        BaseObject extensionVersionObject = getExtensionVersionObject(document, extensionVersion, false);
```

Comment:
I checked it all - and it already worked optimally. I wrote some javadoc comments for bigger clarity